src/styles/blocks/product-card.scss

```diff
@@ -0,0 +1,108 @@
+@import "/styles/utils/reset.scss";
+@import "/styles/utils/_variables.scss";
+@import "/styles/blocks/stars.scss";
+
+.catalog {
+  width: $catalog-width;
+  display: flex;
+  flex-wrap: wrap;
+  padding: 48px 38px;
+  margin: 0 auto;
+}
+
+.container {
+  width: $card-width;
+  margin-right: $card-margin;
+  &:nth-child(4n) {
+    margin-right: 0;
+  }
+  &:nth-child(-n+4) {
+    margin-bottom: 48px;
+  }
+}
+
+.product-card {
+  display: flex;
+  padding: 32px 16px 16px;
+  flex-direction: column;
+  border: 1px solid #f3f3f3;
+  border-radius: 5px;
+
+  &__rate,
+  &__price {
+    display: flex;
+    justify-content: space-between;
+    align-items: center;
+  }
+  &__rate {
+    margin-bottom: 24px;
+  }
+  &__image,
+  &__name {
+    align-self: center;
+  }
+  &__name {
+    display: flex;
+    font-size: 12px;
+    font-weight: 500;
+    line-height: 18px;
+    color: $primary-color;
+    margin-bottom: 4px;
+  }
+  &__code {
+    font-size: 10px;
+    font-weight: 400;
+    line-height: 14px;
+    color: $secondary-color;
+    margin-bottom: 16px;
+    margin-top: 0;
+  }
+  &__review {
+    font-size: 10px;
+    font-weight: 400;
+    color: $primary-color;
+  }
+  &__title {
+    font-size: 12px;
+    font-weight: 400;
+    color: $secondary-color;
+  }
+  &__value {
+    font-size: 16px;
+    font-weight: 700;
+    color: $primary-color;
+  }
+  &__button {
+    background-color: $button-color;
+    line-height: 38px;
+    width: 100%;
+    border-radius: 5px;
+    color: white;
+    font-size: 14px;
+    font-weight: 700;
+    text-transform: uppercase;
+    text-decoration: none;
+    padding: 0;
+    align-self: center;
+    text-align: center;
+    justify-content: center;
+    border: 1px solid transparent;
+  }
```

Comment:
Add transition to button, remove redundant styles : padding, aligh-self, justify-content
Also, like for me better add inline-block property and use padding to increase clickable space